data/src/test/java/org/apache/iceberg/io/TestTaskEqualityDeltaWriter.java

```diff
@@ -409,6 +421,55 @@ public void testUpsertDataWithFullRowSchema() throws IOException {
         .isEqualTo(ImmutableList.of(posRecord.copy("file_path", dataFile.path(), "pos", 0L)));
   }
 
+  @TestTemplate
+  public void testDeleteFileGranularity() throws IOException {
+    withGranularity(DeleteGranularity.FILE);
+  }
+
+  @TestTemplate
+  public void testDeletePartitionGranularity() throws IOException {
+    withGranularity(DeleteGranularity.PARTITION);
+  }
+
+  private void withGranularity(DeleteGranularity granularity) throws IOException {
+    List<Integer> eqDeleteFieldIds = Lists.newArrayList(idFieldId, dataFieldId);
+    Schema eqDeleteRowSchema = table.schema();
+
+    GenericTaskDeltaWriter deltaWriter =
+        createTaskWriter(eqDeleteFieldIds, eqDeleteRowSchema, granularity);
+
+    Map<Integer, Record> expected = Maps.newHashMapWithExpectedSize(2000);
+    // Create enough records, so we have multiple files
+    for (int i = 0; i < 2000; ++i) {
+      Record record = createRecord(i, "aaa" + i);
+      deltaWriter.write(record);
+      if (i % 5 == 10) {
```

Comment:
this seems wrong

Reply:
Good catch!
Actually found an issue I was expecting, so I needed a new delete writer